simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/StationMessageSender.java

```diff
@@ -267,6 +269,22 @@ public void sendHeartBeatAndSubscribe(HeartbeatRequest heartbeatRequest, Subscri
         sendMessage(new WebSocketClientInboxMessage.OcppMessage(call.toJson()));
     }
 
+    /**
+     * Send NotifyReport event
+     * @param requestId requestId from GetBaseReport
+     * @param tbc to be continued, signifies if this is the last report
+     * @param seqNo sequence number of this message
+     * @param reportData report data containing information about variables
+     */
+    public void sendNotifyReport(@Nullable Integer requestId, boolean tbc, int seqNo, ZonedDateTime generatedAt, List<ReportDatum> reportData) {
```

Comment:
I accept null only here. getRequestId from the request class already returns nullable Integer, I don't see the value of converting it to a primitive integer, since the purpose is to simply pass this field to the response

Reply:
why do you convert other arguments then?

Reply:
I don't I set these, as they are required.

Reply:
I did not go through all of them but: seqNo is not required

Reply:
And also generateAt should be null or not? How about reportData?

Reply:
<img width="940" alt="Screenshot 2019-03-21 at 16 46 04" src="https://user-images.githubusercontent.com/47893389/54764927-eb1c8c00-4bf8-11e9-9965-e3cb1200f61b.png">

Reply:
My mistake true. its required. REgarding non-null arguments please read my comment above